packages/swap/src/utils/getPoolQuote.ts

```diff
@@ -90,14 +106,24 @@ export default async function getPoolQuote({
     destAmount: swapOutputAmount,
   });
 
-  includedFees.push(
-    buildFee(getInternalAsset(ingressFee), 'INGRESS', ingressFee.amount),
-    buildFee('Usdc', 'NETWORK', networkFee.amount),
-  );
-  if (brokerFee.amount > 0n) {
-    includedFees.push(buildFee('Usdc', 'BROKER', brokerFee.amount));
+  includedFees.push({ ...ingressFee, type: 'INGRESS' });
+
+  if (isOnChain && networkFee.amount > env.MINIMUM_NETWORK_FEE_USDC) {
+    const networkFeeBps = isStableCoinSwap(srcAsset, destAsset)
+      ? env.ON_CHAIN_STABLECOIN_NETWORK_FEE_BPS
+      : env.ON_CHAIN_DEFAULT_NETWORK_FEE_BPS;
+    const normalNetworkFeeBps = 10n;
+    networkFee.amount = bigintMax(
+      (networkFee.amount * networkFeeBps) / normalNetworkFeeBps,
+      env.MINIMUM_NETWORK_FEE_USDC,
```

Comment:
why are we dividing here? 🧐

Reply:
also, the output is unaffected right? we gonna wait for protocol to change swap_rate_v3 ?

Reply:
i meant as in the egress amount has not been adjusted to account for the new network fees?